src/printer.js

```diff
@@ -4089,11 +4089,7 @@ function printBinaryishExpressions(
     // precedence level and should be treated as a separate group, so
     // print them normally. (This doesn't hold for the `**` operator,
     // which is unique in that it is right-associative.)
-    if (
-      util.getPrecedence(node.left.operator) ===
-        util.getPrecedence(node.operator) &&
-      node.operator !== "**"
-    ) {
+    if (util.shouldFlatten(node.operator, node.left.operator)) {
```

Comment:
Changing this was required due to `needsParens` not actually being called on the to-be-flattened expression.